src/Spanner/Database.php

```diff
@@ -0,0 +1,1313 @@
+
+    /**
+     * @var Instance
+     */
+    private $instance;
+
+    /**
+     * @var LongRunningConnectionInterface
+     */
+    private $lroConnection;
+
+    /**
+     * @var Operation
+     */
+    private $operation;
+
+    /**
+     * @var string
+     */
+    private $projectId;
+
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var Iam
+     */
+    private $iam;
+
+    /**
+     * @var Session|null
+     */
+    private $session;
+
+    /**
+     * @var SessionPoolInterface|null
+     */
+    private $sessionPool;
+
+    /**
+     * Create an object representing a Database.
+     *
+     * @param ConnectionInterface $connection The connection to the
+     *        Google Cloud Spanner Admin API.
+     * @param Instance $instance The instance in which the database exists.
+     * @param LongRunningConnectionInterface $lroConnection An implementation
+     *        mapping to methods which handle LRO resolution in the service.
+     * @param string $projectId The project ID.
+     * @param string $name The database name.
+     * @param SessionPoolInterface $sessionPool [optional] The session pool
+     *        implementation.
+     * @param bool $returnInt64AsObject [optional If true, 64 bit integers will
+     *        be returned as a {@see Google\Cloud\Core\Int64} object for 32 bit
+     *        platform compatibility. **Defaults to** false.
+     */
+    public function __construct(
+        ConnectionInterface $connection,
+        Instance $instance,
+        LongRunningConnectionInterface $lroConnection,
+        array $lroCallables,
+        $projectId,
+        $name,
+        SessionPoolInterface $sessionPool = null,
+        $returnInt64AsObject = false
+    ) {
+        $this->connection = $connection;
+        $this->instance = $instance;
+        $this->lroConnection = $lroConnection;
+        $this->lroCallables = $lroCallables;
+        $this->projectId = $projectId;
+        $this->name = $name;
+        $this->sessionPool = $sessionPool;
+        $this->operation = new Operation($connection, $returnInt64AsObject);
+
+        if ($this->sessionPool) {
+            $this->sessionPool->setDatabase($this);
+        }
+    }
+
+    /**
+     * Return the simple database name.
+     *
+     * Example:
+     * ```
+     * $name = $database->name();
+     * ```
+     *
+     * @return string
+     */
+    public function name()
+    {
+        return $this->name;
+    }
+
+    /**
+     * Check if the database exists.
+     *
+     * This method sends a service request.
+     *
+     * **NOTE**: Requires `https://www.googleapis.com/auth/spanner.admin` scope.
+     *
+     * Example:
+     * ```
+     * if ($database->exists()) {
+     *     echo 'Database exists!';
+     * }
+     * ```
+     *
+     * @param array $options [optional] Configuration options.
+     * @return bool
+     */
+    public function exists(array $options = [])
+    {
+        try {
+            $this->ddl($options);
```

Comment:
why call ddl here instead of GetDatabase:
https://cloud.google.com/spanner/docs/reference/rpc/google.spanner.admin.database.v1#google.spanner.admin.database.v1.DatabaseAdmin.GetDatabase